src/main/java/ru/mail/polis/homework/simple/HomeworkTask.java

```diff
@@ -10,32 +12,56 @@ public class HomeworkTask {
      * Считаем, что функция определена на всем пространстве от a до b
      */
     public static double calcIntegral(double a, double b, ToDoubleFunction<Double> function, double delta) {
-        return 0;
+        BigDecimal sumAccumulator = new BigDecimal(0);
+        for (double i = a; i <= b; i += delta) {
+            sumAccumulator = sumAccumulator.add(BigDecimal.valueOf(function.applyAsDouble(i) * delta));
+        }
+        return sumAccumulator.doubleValue();
     }
 
     /**
      * Вывести номер максимальной цифры. Счет начинается слева направо,
      * выводим номер первой максимальной цифры (если их несколько)
      */
     public static byte maxNumber(long a) {
-        return 0;
+        String aStr = String.valueOf(a);
+        boolean isAPositive = a >= 0;
+        int aMax = 0;
+        while (a != 0) {
+            aMax = (int) Math.max(aMax, Math.abs(a % 10));
+            a /= 10;
+        }
+        if (isAPositive) {
+            return (byte) (aStr.indexOf(String.valueOf(aMax)) + 1);
+        }
+        return (byte) (aStr.indexOf(String.valueOf(aMax)));
     }
 
-
     /**
      * Даны две точки в пространстве (x1, y1) и (x2, y2). Вам нужно найти Y координату третьей точки (x3, y3),
      * которая находится на той же прямой что и первые две.
      */
     public static double lineFunction(int x1, int y1, int x2, int y2, int x3) {
-        return 0;
+        return (x3 - x1) * (y2 - y1) / (x2 - x1) + y1;
```

Comment:
С плавающей запятой при делении проблема была скорее всего, вот это работает у меня:

```suggestion
        return (x3 - x1) * (y2 - y1) / (double) (x2 - x1) + y1;
```